### PR TITLE
Initial swift support

### PR DESCRIPTION
**Pull request by rudolphpienaar**
Hi all -- this PR shouldn't affect current pfcon.  It adds some JSON to internal var for swift client connection, as well as some methods to handle swift put_object. Right now, the code will not execute the swift parts (turned off with an ``and False`` in line 1134, so there should be no impact on pfcon images.

I'd like to commit so that I can continue debugging/developing -- the pfcon docker image does not have the python-swiftclient, so if I can do this PR that should add it.

I'll test the e2e and email again.

### CODE REVIEW

**Comment by rudolphpienaar** [comment]
OK -- finished testing. I had to use local builds of containers. e2e on ChRIS/CUBE worked -- but the auto builds fail in this case due to an edge case. It seems that the CUBE_test.sh uses an older pfcon image? Or at least one that does not have the python-swiftclient installed.

@danmcp  any ideas/suggestions? I'm reasonably sure this PR won't break things... Maybe if you can take a look? We can defer also to tomorrow...

File: pfcon/pfcon.py [hunk 1]
```diff
@@ -44,7 +45,14 @@
         'version':              'undefined',
         'coordBlockSeconds':    10
     },
-
+    "swift": {
+        "auth_url":                 "http://swift_service:8080/auth/v1.0",
+        "username":                 "chris:chris1234",
```
Comment (danmcp) [inline pfcon/pfcon.py]: Do you really want to put this in pfcon.py and not require it to come from config?

Comment (rudolphpienaar) [reply pfcon/pfcon.py]: I initially wanted to, but that would have meant an extra "step" in the e2e setup -- i.e. a step where once pfcon is up, that it is configured. I guess I could add this logic to CUBE -- CUBE knows about swift and has these parameters. It also has the curl client pfurl so in theory CUBE could configure pfcon.

Maybe this is a better idea.

**Comment by danmcp** [comment]
@rudolphpienaar Please squash the commits.

**Comment by rudolphpienaar** [comment]
@danmcp @ravisantoshgudimetla @jbernal0019 

I'd like to merge this PR. Can you look and comment? Once this is fine and pulled into master, I can properly do the CUBE e2e on travis. These currently fail since the CUBE e2e is using the wrong pfcon image (it is using one w/o swift support).

**Comment by danmcp** [comment]
@rudolphpienaar Squashing the commits would be ideal

File: swiftCtl.sh [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+
+        for FILE in $LIST ; do
+              printf "Pulling file: %s\n" $FILE
+              CMD="swift -U chris:chris1234 -A http://${SWIFTIP}:${SWIFTPORT}/auth/v1.0  -K testing download users $OUTPUTDIR --prefix $FILE"
+              CMD_eval "$CMD"
+        done
+        ;;
+esac
+
+# if (( Gb_verbose )) ; then
```
Comment (danmcp) [inline swiftCtl.sh]: Is there a reason to check in the commented out code?

Comment (rudolphpienaar) [reply swiftCtl.sh]: Good catch. No, there isn't

**Pull request merged**